### ISSUE TITLE
@mentions in the main review thread do not get marked as mentions

### ISSUE THREAD

**Issue by nate-tri** [Reviewable/Reviewable#636]
If I @mention someone in the main review discussion thread on Reviewable, the PR does not show up in the Pull Request mentions filter.  Thus they also don't show up in the "Involving Me:" listing in the reviewable dashboard for that person.

If I @mention someone in the comment stream directly in GitHub (enterprise), then the PR will show up in the Pull Request mentions filter and thus will show up in the "Involving Me:" dashboard.

Reviewable v2056.3350
GitHub Enterprise 2.13.22

**Comment by pkaminski** [comment]
Hey Nate, what happens if you @-mention someone in a review comment on GitHub rather than the issue's main thread?  This would be the closest equivalent for the kinds of comments that Reviewable posts via the API.

**Comment by pkaminski** [comment]
Thanks, reproduced on github.com and query sent to GitHub support, as this seems to either be a bug in their system or a basic misunderstanding on my part on how @-mentions should work.  I'll follow up here when I have more information.

**Comment by pkaminski** [comment]
GitHub support confirmed this is a bug on their side and escalated to the engineering team.  No workarounds suggested (or likely possible) and no ETA given for a fix.

**Comment by nate-tri** [comment]
Circling back on this. Any updates, and/or is there a Github issue link that you can share here so we can add our :fire: +1 and perhaps push on it via our own internal GHE support channels?

**Comment by pkaminski** [comment]
I just pinged GitHub support again about this and they haven't done any work on the bug at all.  Expressing interest in a fix could help since they prioritize by impact.  My original GH support ticket is 132327 in case that helps them aggregate reports.

**Comment by nate-tri** [comment]
Can we perhaps apply a "hack" where an @mention in the main thread triggers reviewable to create a separate top level discussion with the @mention in it?  It's a little gross, but we're getting bitten by this a lot and GitHub doesn't seem to be fixing it at all.

**Comment by pkaminski** [comment]
I intend to take a swing at #437 soon, which would alleviate the issue by posting line comments as individual review comments.  Top-level mentions would still be a problem but perhaps a small enough one to either ignore or employ the workaround you suggest...

**Comment by pkaminski** [comment]
I added a workaround for this issue.  Reviewable will now manually index @-mentions in GH reviews and add those PRs to the dashboard.  This works best for connected repos.  Mentions made prior to this fix (or in unconnected repos) are backfilled lazily if/when the review gets synced.  PRs that are closed and later reopened may not reappear on the list unless the user manually extends the concluded reviews horizon to include the time they were closed ("Show more concluded reviews" link).
